ChaosMod/Components/EffectDispatcher.cpp

```diff
@@ -566,6 +569,16 @@ void EffectDispatcher::ClearMostRecentEffect()
 	}
 }
 
+std::vector<RegisteredEffect*> EffectDispatcher::GetRecentEffects(int distance)
+{
+	std::vector<RegisteredEffect*> temp;
+	for (int i = m_rgDispatchedEffectsLog.size() - 1; i --> distance;)
+	{
+		temp.emplace_back(m_rgDispatchedEffectsLog.at(i));
```

Comment:
Maybe it should check if an effect is already in the list and ignore it (as you can't have 2 instances of an effect running simultaneously).